### ISSUE TITLE
No response data returned when using download

### ISSUE THREAD

**Issue by issunboshi** [barryvdh/laravel-snappy#264]
Hi,

First of all, this plugin on the whole is great. The issue I'm having may be implementation but I can't find any consistency between PDFs which download correctly and those which return no response data.

I'm using Laravel v5.4.36 and v0.4.0 of Snappy.

I'm using Angular 5 on the front-end. Any requests for PDFs are setting responseType:3. The issue isn't how Angular interprets the data it's that no data is returned when generating some PDFs.

Here are things I've tried to get something to return when using `$pdf->download` or `$pdf->stream`:

- The complexity of the PDF HTML doesn't have any bearing. I can try outputting a simple `<h1>Hello World</h1>` and I get the same result (nothing) that I would from outputting my multi-page HTML
- If I look at the HTML being generated it all looks fine
- If I try to 'save()' the PDFs it is generated correctly and I can view it without issue
- I've tried using both `PDF::loadView('pdfs.reminder', compact('transactions', 'globalData'))` and `\View::make('pdfs.reminder', compact('transactions', 'globalData'))` to generate the HTML. That doesn't have any effect.

Here is the code I'm currently using:
```
private function generatePDF($transactions, $globalData)
{
		$html = \View::make('pdfs.reminder', compact('transactions', 'globalData'))->render();
		$pdf = \PDF::loadHTML($html);

		return $pdf->download('reminder.pdf');
}
```

It's also worth noting that some PDFs I've tried to generate have worked. But that is a 1/3 success ratio so far. I'm using the same request format and same methods for all 3. No errors are being thrown.

I'm hoping there is something obvious that I'm missing? But I've tried to solve it without success so far. Any help would be appreciated.

Thanks,

Cliff

**Comment by fridzema** [comment]
@issunboshi What happens if you test with `$pdf->inline()'? 
Speaking for myself i've had a couple of times a blank pdf but never an empty page without response.

**Comment by issunboshi** [comment]
Hi @fridzema Thanks for replying so quickly. I've just tried `$pdf->inline()` but I still receive nothing back. Something very strange is happening.

**Comment by fridzema** [comment]
@issunboshi Do you get nothing back at angular or at the browser? 
What happens if you load the url manually in the browser?

**Comment by issunboshi** [comment]
Hi @fridzema. I received nothing back when looking at the request and response in the Network tab in Chrome.

My request is a POST request so couldn't be done manually in the browser. However, I've just setup a simple GET version that used some hardcoded values and visiting it in the browser does generate the PDF correctly. There isn't be a reason why a POST request wouldn't be able to trigger the download, is there?

**Comment by fridzema** [comment]
@issunboshi Hmmm thats strange i really can't answer that, i am currently only using it with a simple GET request. Maybe you can find some info here https://stackoverflow.com/questions/25575620/post-request-to-laravel-then-download-string-as-a-file

**Comment by issunboshi** [comment]
Hi @fridzema, I eventually got the bottom of this. When I called `$this->generatePDF()` I wasn't `return`ing anything, such a basic error. There's no issue with responding with a download from a POST request, so long as the client side code handles it in some way (which mine was handling) I just failed to send a response. Thanks for your help, you led me to look at how I was calling various methods and then I noticed the problem.
